### ISSUE TITLE
Improve dune as a flake support

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/dune#7192]
As discussed on slack, dune is quite heavy when is used as a nix flake in other projects. This is due to opam2nix, which is useful for developers but adds unnecessary weight to plain nix users.

**Comment by Alizter** [comment]
After gaining some more experience with flakes, I think I could rewrite it to avoid opam2nix.

**Comment by rizo** [comment]
@rgrinberg just a quick question: when you say "heavy" are you referring to the dependency closure or something else?

**Comment by rgrinberg** [comment]
Yes, I just mean the evaluation of the opam2nix scope which requires running the solver.
